docs/proposals/error-handling-guideline.md

```diff
@@ -0,0 +1,221 @@
+# ORAS CLI Error Handling and Message Guidelines
+
+This document aims to provide the guidelines for ORAS contributors to improve existing error messages and error handling method as well as the new error output format. It will also provide recommendations and examples for ORAS CLI contributors for how to write friendly and standard error messages, avoid generating inconsistent and ambiguous error messages.
+
+## General guiding principles
+
+A clear and actionable error message is very important when raising an error, so make sure your error message describes clearly what the error is and tells users what they need to do if possible.
+
+First and foremost, make the error messages descriptive and informative. Error messages are expected to be helpful to troubleshoot where the user has done something wrong and the program is guiding them in the right direction. A great error message is recommended to contain the following elements:
+
```

Comment:
Agree. Error code is optional and is generally from the server side. So in some cases error code is also necessary to be showed in the error message.
Splitting "What the error is" and "How to solve it" into two lines looks much straightforward for users. If you look at Docker, Docker error messages are also following this style.

Reply:
How will the error title be used? Giving a usage example would clarify this.

Reply:
I removed "error title" because this is a bit redundant. A concise and brief error description is enough.
 
The recommended error message structure is listed at [line49](https://github.com/oras-project/oras/pull/1163/files#diff-87978b235bb041f07bf2e71966c261dd6103d718e95ebb9c5c984bebbaf5d357R49).